cost-analyzer/templates/forecasting-deployment.yaml

```
@@ -0,0 +1,102 @@
+{{- if .Values.forecasting.enabled }}
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ template "forecasting.fullname" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{ include "forecasting.commonLabels" . | nindent 4 }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      {{ include "forecasting.selectorLabels" . | nindent 6 }}
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: forecasting
+        app.kubernetes.io/instance: {{ .Release.Name }}
+        app: forecasting
+      {{- with .Values.global.podAnnotations}}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+    spec:
+      restartPolicy: Always
+      containers:
+        - name: forecasting
+          {{- if .Values.kubecostModel }}
+          {{- if .Values.kubecostModel.openSourceOnly }}
+          {{- fail "Kubecost Forecasting cannot be used with open source only" }}
+          {{- else if .Values.forecasting.fullImageName }}
+          image: {{ .Values.forecasting.fullImageName }}
+          {{- else if .Values.kubecostModel.fullImageName }}
+          image: {{ .Values.kubecostModel.fullImageName }}
+          {{- else if .Values.imageVersion }}
+          image: {{ .Values.kubecostModel.image }}:{{ .Values.imageVersion }}
+          {{- else }}
+          image: {{ .Values.kubecostModel.image }}:prod-{{ $.Chart.AppVersion }}
+          {{ end }}
+          {{- else }}
+          image: gcr.io/kubecost1/cost-model:prod-{{ $.Chart.AppVersion }}
+          {{ end }}
+          args: ["forecasting"]
+          {{/*
+          fix readinessProbe
+
+          readinessProbe:
+            httpGet:
+              path: /healthz
+              port: 5000
+            initialDelaySeconds: 10
+            periodSeconds: 5
+            failureThreshold: 200
+          */}}
+          imagePullPolicy: Always
+          ports:
+            - name: tcp-api
+              containerPort: 5000
+              protocol: TCP
+          resources:
+            {{- toYaml .Values.forecasting.resources | nindent 12 }}
+          volumeMounts:
+          env:
+            - name: CONFIG_PATH
+              value: /var/configs/
+            - name: KCM_BASE_URL
+              value: "https://aggregator.nightly.kubecost.xyz/model"
```

```suggestion
              value: {{ template "cost-analyzer.serviceName" . }}
```

This _should_ work. This will point to the k8s DNS name of the `cost-analyzer` service. You may need to add `:9003/model` as well.

---

Yes, this is the right thing to do. As Thomas said, you may need to add a port and route -- _not_ 9003 though! `:9090/model`, or `:9004` if you want to hit aggregator directly. 9003 is the cost-model container.

---

Oops! Great catch thank you